pa-rserve/src/main/java/org/ow2/parserve/PARServeEngine.java

```diff
@@ -0,0 +1,446 @@
+        return new RServeConf(null, rServePort, login, password, timeout, daemon, debug, rServeProperties, rEnvProperties);
+    }
+
+    /**
+     * This method initialize PTK env variables on Rserve
+     *
+     * @throws REXPMismatchException
+     * @throws REngineException
+     */
+    private void initializePTK() throws REXPMismatchException, REngineException {
+
+        // deactivate Node process tree killer (by replacing the existing one), to avoid duplicate killing
+        final CookieBasedProcessTreeKiller processKillerDeactivated = CookieBasedProcessTreeKiller.createAllChildrenKiller(
+                "node");
+        final CookieBasedProcessTreeKiller processKiller = CookieBasedProcessTreeKiller.createAllChildrenKiller(
+                "_RServe");
+
+        Rsession initSession = Rsession.newInstanceTry(PARServeEngine.class.getSimpleName(), rServeConf);
+
+        initSession.eval("Sys.setenv(" + processKiller.getCookieName() + "=\"" + processKiller.getCookieValue() + "\")");
+        final int rServePid = initSession.eval("Sys.getpid()").asInteger();
+
+        logger.info("Rserve PID : " + rServePid);
+
+        initSession.end();
+
+        Runtime.getRuntime().addShutdownHook(new Thread() {
+            @Override
+            public void run() {
+                logger.info("Killing PARServe.");
+                if (OperatingSystem.getOperatingSystem().equals(OperatingSystem.windows)) {
+                    try {
+                        // on windows, PTK does not seem to work for RServe, rely on taskkill command.
+                        Runtime.getRuntime().exec(new String[]{"taskkill", "/F", "/PID", "" + rServePid, "/T"}).waitFor();
+                    } catch (Exception e) {
+                        logger.error(e);
+                    }
+                } else {
+                    processKiller.kill();
+                }
+                logger.info("PARServe killed.");
+            }
+        });
+    }
+
+    @Override
+    public Object eval(String script, ScriptContext ctx) throws ScriptException {
+        // Transfer all bindings from context into the rengine env
+        if (ctx == null) {
+            throw new ScriptException("No script context specified");
+        }
+        Bindings bindings = ctx.getBindings(ScriptContext.ENGINE_SCOPE);
+        if (bindings == null) {
+            throw new ScriptException("No bindings specified in the script context");
+        }
+
+        serverEval = false;
+        Map<String, Serializable> variables = (Map<String, Serializable>) bindings.get(TASK_SCRIPT_VARIABLES);
+        if (variables != null) {
+            serverEval = "true".equals(variables.get(PARSERVE_SERVEREVAL));
+        }
+
+
+        engine = new PARServeConnection(Rsession.newInstanceTry("Script", rServeConf), serverEval);
+
+        initializeTailer(bindings, ctx);
+
+        Object resultValue = null;
+
+        try {
+
+            if (!serverEval) {
+                this.enableWarnings(ctx);
```

Comment:
All instructions inside the _if_ block seems to be the same as in PAJRIEngine. I think it could be factorized by put the code (all assign calls) into a method declared in the parent class (PAREngine)?